packages/components/src/FilterBar/subcomponents/FilterSelect/FilterSelect.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react"
+import { FilterButton } from "~components/FilterButton"
+import {
+  FilterSelect,
+  FilterSelectProps,
+  SelectOption,
+} from "~components/FilterSelect"
+import { useFilterBarContext } from "../../context/FilterBarContext"
+
+export type FilterBarSelectProps<Option extends SelectOption = SelectOption> =
+  Omit<
+    FilterSelectProps<Option>,
+    "isOpen" | "setIsOpen" | "renderTrigger" | "label" | "selectedKey"
+  > & {
+    id?: string
+  }
+
+export const FilterBarSelect = <Option extends SelectOption = SelectOption>({
+  id,
+  onSelectionChange,
+  ...props
+}: FilterBarSelectProps<Option>): JSX.Element => {
+  const { getFilterState, toggleOpenFilter, updateValue } =
+    useFilterBarContext<Option["value"]>()
+
+  if (!id) throw Error("Missing `id` prop in FilterBarSelect")
+
+  const filterState = getFilterState(id)
```

Comment:
Thoughts on sending the `id` directly to `useFilterBarContext`? Or perhaps rename `useFilterBarContext` to `useFilterState`?

Reply:
Renaming the hook isn't a good idea because it's a context, and you don't want anyone confusing it to be similar to a `useState`. More util functions will be coming out of the context later, so even if it looks like it matches now, it won't later.

Sending the `id` to the context hook probably wouldn't work, since it's a context consumer. Well, I suppose it _could_ work, but it means we'd need to have it call all the functions provided by the provider, so essentially it's an almost-duplicate 🤔
The problem is, that therefore ties all the functions to the provided `id`, and if you wanted to get values from another filter (eg. a Dependent Filter wants to check that a Source Filter has a value, and then it will open itself), you can't, unless I also provide you the original functions at the same time 😅

I'm not sure if it's worth the added complexity given the usage isn't that complicated.

Reply:
Fair enough!